### PR TITLE
[feat] #4 - mnist dataset을 활용한 3층 신경망 학습 구현

### LINKED ISSUE #4: [feat] mnist dataset을 활용한 3층 신경망 학습 구현
## Task
---
- mnist dataset을 활용한 3층 신경망 학습 기능을 구현합니다.

## TO-DO
---
- [x] softmax function 구현
- [x] 배치처리 구현
- [x] backpropagation 구현
- [x] activation function 구현
- [x] loss function 구현

### PR DESCRIPTION
**Pull request by Bbuterfly**
# Related Issue
- Resolved: #4

# Work Description
- mnist dataset을 활용한 3층 신경망 학습 구현

# Checklists
- [x] 필요없는 주석, 프린트문 제거했는지 확인
- [x] 컨벤션 지켰는지 확인

### CODE REVIEW

**Comment by Bbuterfly** [comment]
학습 코드는 돌아가기는 하지만 accracy가 0.1(무작위로 찍음)에서 학습되지 않는 문제가 있습니다.
이는 다양한 문제점이 있겠지만 gradient 소실/폭발, dead Relu, 초기화문제의 가능성이 있어보입니다.
또한 optimize 방식을 GD 뿐 아니라 SGD, Momentum, Adagrad, Adam등을 구현해보는 것도 좋아보입니다.
